### DISCUSSION TITLE
Job级别的重试可以触发后续依赖Job的重试

### DISCUSSION THREAD

**Discussion by GJieHuang** [Tencent/bk-ci#4269] [Ideas]
**What would you like to be added**:
同一个Stage下，存在依赖的Job，在前置Job失败重试后， 后续依赖的Job也会自动重试
eg. 某个Stage 下有1、2、3 三个job ， 3依赖于2依赖于1
1失败后，点击此Job下的重试，无法触发后续的2、3执行
导致此次构建作废，只能重新构建

**Why is this needed**:
虽然点击Stage里的重试可以触发1、2、3全部重新执行
但若误点了1的重试，此次构建就作废了

**Comment by zanyzhao** [comment]
> 但若误点了1的重试，此次构建就作废了

你说的1的重试是指？

**Reply by GJieHuang** [reply]
job1 失败后显示的重试。
或者直接把job层级的重试按钮去掉，直接通过Stage级别的重试按钮操作
避免误操作
![image](https://user-images.githubusercontent.com/20061612/119824147-2aa53800-bf28-11eb-9cc2-e8a3dbe56301.png)

**Reply by zanyzhao** [reply]
- https://github.com/Tencent/bk-ci/issues/4201#issuecomment-844630379
- #2175 

后续可能会针对这两个issue做一些调整，比如：
1. 禁止在编译环境里执行单插件重试
2. 无编译环境下的插件设置里增加“是否允许重试”的开关

**Reply by iccengan** [reply]
#5175 跟进
